sqldb/sqlc/queries/invoices.sql

```diff
@@ -0,0 +1,197 @@
+-- name: InsertInvoice :one
+INSERT INTO invoices (
+    hash, preimage, memo, amount_msat, cltv_delta, expiry, payment_addr, 
+    payment_request, state, amount_paid_msat, is_amp, is_hodl, is_keysend, 
+    created_at
+) VALUES (
+    $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14
+) RETURNING id;
+
+-- name: InsertInvoiceFeature :exec
+INSERT INTO invoice_features (
+    invoice_id, feature
+) VALUES (
+    $1, $2
+);
+
+-- name: GetInvoiceFeatures :many
+SELECT *
+FROM invoice_features
+WHERE invoice_id = $1;
+
+-- name: DeleteInvoiceFeatures :exec
+DELETE 
+FROM invoice_features
+WHERE invoice_id = $1;
+
+-- This method may return more than one invoice if filter using multiple fields
+-- from different invoices. It is the caller's responsibility to ensure that 
+-- we bubble up an error in those cases.
+
+-- name: GetInvoice :many
+SELECT *
+FROM invoices
+WHERE (
+    id = sqlc.narg('add_index') OR 
+    sqlc.narg('add_index') IS NULL
+) AND (
+    hash = sqlc.narg('hash') OR 
+    sqlc.narg('hash') IS NULL
+) AND (
+    preimage = sqlc.narg('preimage') OR 
+    sqlc.narg('preimage') IS NULL
+) AND (
+    payment_addr = sqlc.narg('payment_addr') OR 
+    sqlc.narg('payment_addr') IS NULL
+)
+LIMIT 2;
+
+-- name: FilterInvoices :many
+SELECT * 
+FROM invoices
+WHERE (
+    id >= sqlc.narg('add_index_get') OR 
+    sqlc.narg('add_index_get') IS NULL
+) AND (
+    id <= sqlc.narg('add_index_let') OR 
+    sqlc.narg('add_index_let') IS NULL
+) AND (
+    state = sqlc.narg('state') OR 
+    sqlc.narg('state') IS NULL
+) AND (
+    created_at >= sqlc.narg('created_after') OR
+    sqlc.narg('created_after') IS NULL
+) AND (
+    created_at <= sqlc.narg('created_before') OR 
+    sqlc.narg('created_before') IS NULL
+) AND (
+    CASE
+        WHEN sqlc.narg('pending_only')=TRUE THEN (state = 0 OR state = 3)
```

Comment:
nit: why not just match on a passed in state value and do the "pending only" query from code?

Reply:
sgtm